docs/enhancements/crypto-api.md

```diff
@@ -0,0 +1,246 @@
+# Technical Design Input
+
+# Hyperconverged Kubevirt cluster-wide Crypto Policy Setting
+
+## Abstract
+
+Starting from OCP/OKD 4.6, a [cluster-wide API](https://github.com/openshift/enhancements/blob/master/enhancements/kube-apiserver/tls-config.md) is available for cluster administrators to set TLS profiles for OCP/OKD core components. HCO, as an OCP/OKD layered product, will follow along OCP/OKD crypto policy cluster-wide setting, and use the same profile configured for the cluster’s control plane. Configuration of a TLS security profile ensures that OCP/OKD, as well as HCO and its sibling operators and operands, use cryptographic libraries that do not allow known insecure protocols, ciphers, or algorithms.
```

Comment:
I am not aware of any "Externally Facing Services" in the KubeVirt ecosystem so I'm struggling to understand why we should support this.  None of our components are directly exposed to clients outside the cluster.

APIServers (virt-api, cdi-apiserver) are proxied through via kube-apiserver.

Same with any webhooks

cdi-uploaaproxy is expected to be exposed via a Route/Ingress

So exactly what KubeVirt components are  directly exposed out the cluster?

Reply:
I don't see anywhere in the document where these settings are expected to be propagated down to subcomponents.

Reply:
According to:
https://docs.openshift.com/container-platform/4.9/security/tls-security-profiles.html#tls-profiles-kubernetes-configuring_tls-security-profiles ,
setting the TLS security profile in the APIServer CR propagates the setting to a few OpenShift control plane components:
- Kubernetes API server
- Kubernetes controller manager
- Kubernetes scheduler
- OpenShift API server
- OpenShift OAuth API server
- OpenShift OAuth server
- etcd

This whole proposal is just about having the same setting propagated also to the whole Kubevirt Hyperconverged control plane.

Reply:
If I understand @mhenriks correctly, this proposal is trying to be holier than OKD. There, the purpose is to harden crypto policy for extern services. We have no such services; the purpose of this proposal is to let a cluster admin to harden KubeVirt-internal APIs. Do you know, @tiraboschi, if other complex OLM-controlled operators do that? Do internal OKD operators do that?

Reply:
OCP/OKD are aiming to have a configuration option to let the user configure ciphers that would apply to all OpenShift components both for internal and external traffic.
I'm not aware of any OLM-controlled operator that already implemented it, while between CVO-controlled ones CMO already adopted it ( https://github.com/openshift/cluster-monitoring-operator/pull/1380 ) and MCO is currently working in that direction.

Reply:
Good to know. I'd add this background to your proposal. Also, I'd reduce the emphasis on OKD.
We use encryption internally because cluster owners do not trust all workload. Some cluster owners are more suspicious than others and would like to use stronger, more expensive and less compatible crypto. I think that this proposal should be about letting them do that. Just like I can set `/etc/crypto-policies/config` on my laptop, affecting all polite applications, I would like to set my crypto policy once in hco (or okd) and have all components comply.